src/txmempool.h

```diff
@@ -698,6 +702,28 @@ class CTxMemPool
 
     size_t DynamicMemoryUsage() const;
 
+    /** Adds a transaction to the unbroadcast set */
+    inline void AddUnbroadcastTx(const uint256& txid) {
+        LOCK(cs);
+        m_unbroadcast_txids.insert(txid);
+    }
+
+    /** Removes a transaction from the unbroadcast set */
+    inline void RemoveUnbroadcastTx(const uint256& txid, const bool unchecked = false) {
```

Comment:
Any reason why these are inline? I think we don't that elsewhere.

Also, it might be good to move the implementation of this one to the cpp file, so that logging doesn't have to be included in the header and compiled on every include.

Reply:
thought it was more efficient since these functions are so simple. 

we don't have inline functions in txmempool, but there are plenty scattered around the codebase. 

👍 RE logging. I can def move this function to the cpp. Do you prefer I move them all?

Reply:
The others can stay, but I think the `inline` keyword is still not needed? (With "inline" I meant "`inline`" (the keyword))

Reply:
oh interesting.. reading around, seems like `inline` keyword is just a compiler hint & usually the compiler will deduce when necessary? the code compiles fine when I remove them. is there any harm in leaving it in? on the flip, is there any time they *should* be used?

Reply:
@amitiuttarwar `inline` has two independent effects:
* A hint to the compiler that it should try inlining the function (both in C and C++)
* Indicating that this symbol may appear in multiple modules (but they will all be identical). This is only in C++, and makes it possible to have code in header files (as it will end up in every module that includes it). Member functions defined in classes *always* have this implicitly, no need to add the "inline" keyword for it (unless you also want to hinting effect above).

Reply:
ah okay, this is very helpful. I think I'm understanding- its a clue to the compiler to try expanding in line, as well as an indicator that the symbol defined in more than one translation unit are the same, which the compiler can enforce. when a member function is defined, not just declared, within the class definition- the compiler automatically interprets this as an inline member function. so, the keyword `inline` in the implementation I have here is redundant.

Reply:
but, since I'm pulling out the definition of `RemoveUnbroadcastTx` to `txmempool.cpp`, it will no longer be automatically inlined. So, adding `inline` there would be a hint to the compiler to inline the function at the call sites, but the compiler might do it anyway / ignore. 

is this sounding correct?

Reply:
> that the symbol defined in more than one translation unit are the same

Not just that they're the same; that they may occur in multiple translation units in the first place.

Without inline (or a member being defined in a class body), you will get a linker error if you define the same symbol in multiple translation units.

Unless (there's always an unless...), it's in an anonymous namespace as well (or static for globals). In *that* case, every translation unit will get its own individual copy, which won't be seen as conflicting by the linker at all - but they also won't get deduplicated.

Reply:
Once you've moved the function definition to the .cpp file, it can only get inlined anymore inside the txmempool.cpp translation unit (until we enable LTO, which we currently don't do by default or in releases). With or without `inline`, it will still get inlined there, and with or without `inline` it won't get inlined elsewhere. Adding the keyword may still make it a stronger hint to the compiler to do so - unsure.

Reply:
ok ya, got the part where `inline` indicates permission for same symbol in multiple translation units. the part about `inline` in `.cpp` being limited to the file unless the optimization is enabled makes sense. and the special case for anonymous namespace is 🤯 and I'm going to go have to learn / tinker more 😛 

but first I'm going to push fixups. thanks so much for these explanations! I'm learning a lot :)

Reply:
removed `inline` keywords, moved `RemoveUnbroadcastTx` function definition to `txmempool.cpp` & removed including `logging.h` in the header.